lib/block-supports/duotone.php

```diff
@@ -445,15 +442,44 @@ function gutenberg_render_duotone_support( $block_content, $block ) {
 		return $block_content;
 	}
 
-	$colors          = $block['attrs']['style']['color']['duotone'];
-	$filter_key      = is_array( $colors ) ? implode( '-', $colors ) : $colors;
-	$filter_preset   = array(
-		'slug'   => wp_unique_id( sanitize_key( $filter_key . '-' ) ),
-		'colors' => $colors,
-	);
-	$filter_property = gutenberg_get_duotone_filter_property( $filter_preset );
-	$filter_id       = gutenberg_get_duotone_filter_id( $filter_preset );
+	// Possible values for duotone attribute:
+	// 1. array('#000000', '#ffffff').
+	// 2. 'preset-slug'.
+	// 3. 'unset' (remove filter).
+	$duotone_attr = $block['attrs']['style']['color']['duotone'];
+
+	$is_duotone_colors_array = is_array( $duotone_attr ) && count( $duotone_attr ) === 2;
+	$is_duotone_unset        = 'unset' === $duotone_attr;
+	$is_duotone_preset       = ! $is_duotone_colors_array && ! $is_duotone_unset;
+
+	if ( $is_duotone_preset ) {
+		$filter_preset   = array(
+			'slug' => $duotone_attr,
+		);
+
+		// Utilise existing CSS custom property.
+		$filter_property = "var(--wp--preset--duotone--$duotone_attr)";
+	} else {
+		// Handle when Duotone is either:
+		// - "unset"
+		// - an array of colors.
+
+		// Build a unique slug for the filter based on the array of colors.
+		$filter_key      = is_array( $duotone_attr ) ? implode( '-', $duotone_attr ) : $duotone_attr;
```

Comment:
Could `$is_duotone_colors_array` be used here instead of `is_array( $duotone_attr )` or would the `count( $duotone_attr ) === 2` check earlier get in the way here? Sorry for the naïve questions. Trying to learn about duotone more through these questions.

Reply:
This is not a naïve question—the code isn't very clear about what is going on. 

This `is_array` check was making sure the value isn't `'unset'`. Rather than checking for `'unset'`, I thought it would be more future-proof if other special options needed to be added, but it isn't very clear without a code comment.

Reply:
Thanks for the fix.